x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/LabelFieldProducer.java

```diff
@@ -144,4 +147,24 @@ static class AggregateMetricFieldProducer extends LabelLastValueFieldProducer {
             super(name, new LastValueLabel(metric.name()));
         }
     }
+
+    public static class HistogramLastLabelFieldProducer extends LabelLastValueFieldProducer {
+        HistogramLastLabelFieldProducer(String name) {
+            super(name);
+        }
+
+        @Override
+        public void write(XContentBuilder builder) throws IOException {
+            if (isEmpty() == false) {
+                final HistogramValue histogramValue = (HistogramValue) label.get();
+                final List<Double> values = new ArrayList<>();
```

Comment:
`histogramValue.count()` is not the number of values...it is the count value of each bucket and it can't be used before calling `histogramValue.next()`.

Reply:
Ah, I see. Then my comment doesn't make sense.